### ISSUE TITLE
Handle broken zip created with adm-zip in a nicer way

### ISSUE THREAD

**Issue by diox** [mozilla/addons-linter#1938]
[`adm-zip`](https://www.npmjs.com/package/adm-zip) is a popular node package to generate zip files. Somehow, at least one developer using it end up with a weird zip that the linter can't handle. I've attached [the zip file in question](https://github.com/mozilla/addons-linter/files/1857107/extension.zip) to this issue.

The error being thrown is
```
Error: invalid code lengths set
    at Zlib._handle.onerror (zlib.js:370:17)
```

The linter process terminates when hitting that, and that causes a `Validation was unable to complete successfully due to an unexpected error.` message on the server.

We should handle that in a nicer way - I don't suggest managing to proceed with validation as the zipfile is weird (various linux tools I had also have issues with it, though nothing as bad as the linter) - but at least return a true linter error.

**Comment by muffinresearch** [comment]
@diox should this be an upstream bug in yauzl?

**Comment by diox** [comment]
I'm not sure, I'm not familiar enough with the linter and yazl to figure out if that's expected behavior (meaning we should be doing something on our end to catch this) or not. Ultimately is too low priority for us to spend a lot of time on it IMHO, but a contributor would be welcome to do it.
